exotic/exotic.py

```diff
@@ -1686,6 +1687,13 @@ def main():
         else:
             pDict = userpDict
             CandidatePlanetBool = False
+        # Seed random number generator (for run to run consistency)
+        if exotic_infoDict['random_seed']:
+            log_info(f"Setting random number seed to {exotic_infoDict['random_seed']}")
+        else:
+            exotic_infoDict['random_seed'] = int.from_bytes(hashlib.sha256(f"{pDict['pName']}:{pDict['midT']}".encode()).digest()[0:4], byteorder='little')
+            log_info(f"Generated random number seed {exotic_infoDict['random_seed']}")
+        np.random.seed(exotic_infoDict['random_seed'])
```

Comment:
How to seed the RNG really depends on the underlying algorithm being used. In the case of Numpy, it is PCG64, which requires a 128-bit initial seed but also a 128-bit internal state vector. In this case a 256-bit hash would be sufficient since the `seed` argument is used to generate both values. (Over-/under-seeding is not recommended -- may waste CPU and actually have unintended consequences on the output stream. Just mentioned here -- it's not currently the case.) 

However, to guarantee a specific RNG, it should be [specifically declared when invoking Numpy](https://numpy.org/doc/stable/reference/random/generator.html). Scrolling down the page near the bottom should be code for declaring the RNG. **Note:** I believe the PCG64DXSM has a more even statistical distribution of generated values, and [it is **recommended** for any large numerical sampling algorithm](https://numpy.org/doc/stable/reference/random/upgrading-pcg64.html). That being said, I have yet to see results of any of Numpy's implementations being tested against the [gold-standard DIEHARDER suite](https://webhome.phy.duke.edu/~rgb/General/dieharder.php).

I have not deep-dived on this so I cannot provide specifics on seeding PCG64DXSM, but if statistical regularity is a concern, this appears the way to go.

Reply:
Right - logically, there are two things: the quality of the RNG (which is quite good for PCG64DXSM), and the seed used to drive the resulting deterministic (whether or not it is good or bad - the bits out of the RNG are a deterministic sequence, as defined by the algorithm and the seed.  We're using the same RNG run to run, but the method by which the seed is determined, when not provided, is not controlled nor recorded - when uninitialized, the seed is synthesized (generally from clock time and other entropy sources) ONCE, and the sequence generated from that is determined; when initialized, the same thing happens - the quality of the randomness is a function of the algorithm, not the seed.  No seed is good or bad, given a good RNG - it's only what drives the resulting sequence.

Main point is that an uncontrolled initial seed doesn't make your random sequence any higher quality - it only makes it harder to reproduce from run to run: the entropy of the sequence within the run is purely a function of the algorithm, not the seed.